src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -0,0 +1,533 @@
+   * It is assumed that this will only be used by the ReadThread on its socketExecuter and it checks to 
+   * make sure thats the case.  The ByteBuffer returned is not thread safe.
+   * 
+   * @return A ByteBuffer for the ReadThread to use during its read operations.
+   */
+  protected ByteBuffer getBuffer() {
+    if(! ce.verifyReadThread()) {
+      ce.removeClient(this);
+      throw new IllegalStateException("Only the Client Executers ReadThread can access this function!! Client removed from Executer!");
+    }
+    if(readByteBuffer.remaining() < minAllowedReadBuffer) {
+      readByteBuffer = ByteBuffer.allocate(maxBufferSize*2);
+    }
+    return readByteBuffer;
+  }
+  
+  /**
+   * When a read is read from the socket the size of the read is reported to the client.
+   * This also reports the read size to the statistics of this client.
+   * 
+   * This should be called by the client when the SocketExecuter calls addReadBuffer().
+   * 
+   * @param size The size if the read in bytes from the socket.
+   */
+  protected void addToReadSize(int size) {
+    stats.addRead(size);
+    readBufferSize.addAndGet(size);
+  }
+  
+  /**
+   * When a write is added to the client this is called to report the size of the write
+   * to the client.  This write is added to the buffer size with this call.
+   * 
+   * 
+   * @param size The size of the write in bytes.
+   */
+  protected void addToWriteSize(int size) {
+    writeBufferSize.addAndGet(size);
+  }
+  
+  /**
+   * This should be called when getRead() is called.  This reduces the read buffer by said amount. 
+   * 
+   * @param size The size in bytes that is being removed from the buffers.
+   */
+  protected void removeFromReadSize(int size) {
+    readBufferSize.addAndGet(-size);
+  }
+  
+  /**
+   * This should be called when reduceWrite(int size) is called. This removes the size
+   * from the total buffer size and adds the write to the clients write stats.
+   * 
+   * @param size The amount written to the socket.
+   */
+  protected void removeFromWriteSize(int size) {
+    stats.addWrite(size);
+    writeBufferSize.addAndGet(-size);
+  }
+  
+  /**
+   * 
+   * @return the current size of the ReadBuffer.
+   */
+  public int getReadBufferSize() {
+    return readBufferSize.get();
+  }
+  
+  /**
+   * 
+   * @return the current size of the writeBuffer.
+   */
+  public int getWriteBufferSize() {
+    return writeBufferSize.get();
+  }
+  
+  /**
+   * 
+   * @return the current MaxBuffer size allowed.  The read and write buffer use this independently.
+   */
+  public int getMaxBufferSize() {
+    return maxBufferSize;
+  }
+  
+  /**
+   * This is used by the SocketExecuter to set the ThreadExecuter this client is supposed to use.
+   * The ThreadExecuter works in a single threaded manor to keep all the client operations in order.
+   * 
+   * You should not be overriding this.
```

Comment:
I removed that comment.  Technically it "can be" you just need to be careful.  UDPClient does to give UDPServer visibility into it.